src/components/Card.js

```diff
@@ -73,56 +80,71 @@ export const Card = () => {
         >
           <QRCode
             className="qrcode"
-            style={{ paddingRight: portrait ? '' : '1em' }}
+            style={{
+              paddingRight: portrait ? '' : '1em',
+              height: currentSize + 'px',
+              width: currentSize + 'px',
+            }}
             value={qrvalue}
-            size={175}
+            size={currentSize}
           />
 
-          <div className="inputs">
-            <label>{t('wifi.name')}</label>
-            <textarea
-              id="ssid"
-              type="text"
-              maxLength="32"
-              placeholder={t('wifi.name.placeholder')}
-              autoComplete="off"
-              autoCorrect="off"
-              autoCapitalize="none"
-              spellCheck="false"
-              value={network.ssid}
-              onChange={(e) => setNetwork({ ...network, ssid: e.target.value })}
-            />
-            <label
+          <div>
+            <div
               className={`
+                ${currentSize == smallSize ? 'inputs-small' : false}
```

Comment:
Default to always using the Strict Equality Comparison (`===`). This is preferred as it doesn't do any type conversion. 

More info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

Reply:
Choosing which class to apply based on the current size might be better abstracted into a function. I think that'd make it a lot more obvious what's going on here. Consider the following alternative:

```js
const inputsClassSize = () => {
  switch(size.size) {
    case sizeSmall: 
      return 'inputs-small';
      break;
    case sizeMedium: 
      return 'inputs-medium';
      break;
    ...
  }
}

... 

<div className={`inputs ${inputsClassSize()}`}>
```

Reply:
I implemented it but it doesn't work. That is, the qr code resizing works, but not the fields with the network name and password. I will investigate it. The size does not change. The problem is when I resize I see that no class is added to the div. I can see `class="inputs undefined"`. I tried to do it in such a way that I added the action of calling the `inputsClassSize()` function to the radiobutton (small, medium, large). So far, no result. Besides, I see warning in the console:
`Warning: Failed prop type: Invalid prop 'size' of type 'string' supplied to 'QRCodeCanvas', expected 'number'`

Reply:
> Warning: Failed prop type: Invalid prop 'size' of type 'string' supplied to 'QRCodeCanvas', expected 'number'

This is because the `onChange` callback is currently setting `size.size` to a string (e.target.value) for all 3 sizes. Parse it to a number to resolve:
```js
onChange={(e) => { setSize({ ...size, size: Number(e.target.value) })} }
```

Alternatively, since all 3 size radios aren't relying on anything dynamic you could just hardcode each, which might make it a little more obvious:
```js
onChange={(e) => { setSize({ ...size, size: sizeMedium })} }
```

Reply:
> the qr code resizing works, but not the fields with the network name and password. I will investigate it. The size does not change. 

You're right, I observed the same thing locally. I dug into this a bit and it's probably because react doesn't know to rerender when `size.size` changes. I played around with it and figured out another option for your consideration:
1. Leverage the `useEffect` hook to watch for changes to `size.size`. This means React will invoke the useEffect callback anytime `size.size` is changed.
2. Now we know the size has changed and we need to compute a class to apply to the `.inputs`. Invoke the `inputsClassSize` function because that already switches over the sizes.
3. Refactor each case in `inputsClassSize` to set a class name property in our state, instead of returning a string.
4. Finally, update the `.inputs` classNames to use the stateful `size.class` property

Here's the diff of the above changes. Note this is _one_ way to do this, you might be able to clean it up or figure out a better approach!

```diff
diff --git a/src/components/Card.js b/src/components/Card.js
index e71eb76..cdff1c6 100644
--- a/src/components/Card.js
+++ b/src/components/Card.js
@@ -11,6 +11,7 @@ export const Card = () => {
   const sizeLarge = 275;
   const [size, setSize] = useState({
     size: sizeMedium,
+    class: 'inputs-medium',
   });
   const [network, setNetwork] = useState({
     ssid: '',
@@ -38,12 +39,16 @@ export const Card = () => {
   const inputsClassSize = () => {
     switch (size.size) {
       case sizeSmall:
-        return 'inputs-small';
+        setSize({ ...size, class: 'inputs-small' });
+        break;
       case sizeMedium:
-        return 'inputs-medium';
+        setSize({ ...size, class: 'inputs-medium' });
+        break;
       case sizeLarge:
-        return 'inputs-large';
+        setSize({ ...size, class: 'inputs-large' });
+        break;
       default:
+        setSize({ ...size, class: 'inputs-medium' });
         break;
     }
   };
@@ -71,10 +76,12 @@ export const Card = () => {
       setPortrait(true);
     }
 
+    inputsClassSize();
+
     const ssid = escape(network.ssid);
     const password = escape(network.password);
     setQrvalue(`WIFI:T:${network.encryptionMode};S:${ssid};P:${password};;`);
-  }, [network]);
+  }, [network, size.size]);
 
   return (
     <div>
@@ -101,7 +108,7 @@ export const Card = () => {
             size={size.size}
           />
 
-          <div className={`inputs ${inputsClassSize()}`}>
+          <div className={`inputs ${size.class}`}>
             <label>{t('wifi.name')}</label>
             <textarea
               id="ssid"


```

Reply:
Thank you for your help and you explanation. I was looking for a problem but didn't come to that conclusion. Resize now works properly and PR is updated. 
I hardcoded the values for radiobuttons. It seems to be better, we avoid parsing values from string to number and for that I was able to remove for each radiobutton: `value = sizeXYZ`.
The only drawback of this solution is that `useEffect` performs the operation of setting the size of the value on the card unnecessarily many times, i.e. if someone puts 10 characters in the field, the same value for the size of the elements will be set 10 times. That's why I tried a solution with assigning a function to a radiobutton - one click, one function call. 
But at the moment it seems to me that we achieved the effect we expected and it is cool.